pkg/storage/utils/decomposedfs/xattrs/xattrs.go

```diff
@@ -142,21 +191,48 @@ func Set(filePath string, key string, val string) error {
 }
 
 // SetMultiple allows setting multiple key value pairs at once
-// TODO the changes are protected with an flock
-func SetMultiple(filePath string, attribs map[string]string) error {
+// the changes are protected with an file lock
+// If the file lock can not be acquired the function returns a
+// lock error.
+func SetMultiple(filePath string, attribs map[string]string) (err error) {
 
-	// FIXME: Lock here
+	// h, err := lockedfile.OpenFile(filePath, os.O_WRONLY, 0) // 0? Open File only workn for files ... but we want to lock dirs ... or symlinks
+	// or we append .lock to the file and use https://github.com/gofrs/flock
+	var fileLock *flock.Flock
+	fileLock, err = filelocks.AcquireWriteLock(filePath)
+
+	if err != nil {
+		return errors.Wrap(err, "xattrs: Can not acquire write log")
+	}
+	defer func() {
+		rerr := filelocks.ReleaseLock(fileLock)
+
+		// if err is non nil we do not overwrite that
+		if err == nil {
+			err = rerr
+		}
+	}()
+
+	// error handling: Count if there are errors while setting the attribs.
+	// if there were any, return an error.
+	var xerrs int = 0
+	var xerr error
 	for key, val := range attribs {
-		if err := Set(filePath, key, val); err != nil {
-			return err
+		if xerr = xattr.Set(filePath, key, []byte(val)); xerr != nil {
+			// log
+			xerrs++
 		}
 	}
+	if xerrs > 0 {
+		err = errors.Wrap(xerr, "Failed to set all xattrs")
+	}
 	return nil
```

Comment:
This also needs to be changed
```suggestion
	return err
```